File: haskell/Tribonacci/Tribonacci.hs
module Tribonacci where

last3 xs = [last xs] ++ [last(init xs)] ++ [last(init(init xs))]
sumLast3 xs = sum(last3 xs)

tribonacci :: Num a => (a, a, a) -> Int -> [a]
tribonacci (a, b, c) 0 = []
tribonacci (a, b, c) n = take n (recursive n [a, b, c])

recursive n list
  | n == 0 = list
  | otherwise = recursive (pred n) (list ++ [sumLast3 list])

File: python/Tribonacci/Tribonacci.py
def tribonacci(a, n):
    if(n == 0):
        return []
    if(n == 1):
        return [a[0]]
    else:
        for i in range(1, n - 2):
            a.append(sumLast3(a))
        return a

def sumLast3(ary):
    l = last3(ary)
    acc = 0
    for i in l:
        acc = acc + i
    return acc

def last3(a):
    size = len(a)
    l = []
    for i in range(size - 3, size):
            l.append(a[i])
    return l
